docs/resources/user_role.md

```diff
@@ -0,0 +1,57 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "datadog_user_role Resource - terraform-provider-datadog"
+subcategory: ""
+description: |-
+  Provides a Datadog UserRole resource. This can be used to create and manage Datadog user_role. Note that this resource is in beta and is subject to change.
+---
+
+# datadog_user_role (Resource)
+
+Provides a Datadog UserRole resource. This can be used to create and manage Datadog user_role. Note that this resource is in beta and is subject to change.
+
+## Example Usage
+
+```terraform
+resource "datadog_role" "foo" {
+  name = "foo"
```

Comment:
nit: it's bugged me for a while that we use foo & bar in our examples. Can name the roles closer to reality? Like "Admin" or "Dashboard Writer" or something?